From 4aaea83384df67f75d0a361cd9b47dff888ea8ad Mon Sep 17 00:00:00 2001
From: Olga Gerchikov <gerchiko@microsoft.com>
Date: Tue, 07 Dec 2021 17:47:42 +0000
Subject: [PATCH] Revert "[win] fix problem with black flash just before content window is shown"

This reverts commit c9f0bd6dab7ec93c8b14f179917a12b51eb662c1.

Reason for revert: This change regresses PTP fling scrolling on Surface devices. It makes it not as smooth and stops abruptly. You can see the difference between Canary and Stable builds. Specifically, WS_EX_LAYERED flag breaks the behavior.

Original change's description:
> [win] fix problem with black flash just before content window is shown
>
> LegacyRenderWidgetHostHWND is shown together with RenderWidgetHostViewAura
> window and because it has default style it causing black flash for the
> first paint. Set WS_EX_LAYERED window style in order to
> LegacyRenderWidgetHostHWND window be blend with windows under it.
> Set also WS_EX_NOREDIRECTIONBITMAP flag to avoid additional bitmap in
> layered window.
>
> Bug: 1257540
> Change-Id: Ia9807e694ccb686e0610980bae2febc9f596e8aa
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3212029
> Reviewed-by: Rafael Cintron <rafael.cintron@microsoft.com>
> Reviewed-by: Ken Buchanan <kenrb@chromium.org>
> Commit-Queue: Rafael Cintron <rafael.cintron@microsoft.com>
> Cr-Commit-Position: refs/heads/main@{#940943}

Bug: 1257540
Change-Id: If9ddc311ee14b8fb744359920b5e56b8d36fcefb
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3313944
Reviewed-by: Sunny Sachanandani <sunnyps@chromium.org>
Reviewed-by: Ken Buchanan <kenrb@chromium.org>
Commit-Queue: Sunny Sachanandani <sunnyps@chromium.org>
Commit-Queue: Olga Gerchikov <gerchiko@microsoft.com>
Cr-Commit-Position: refs/heads/main@{#949015}
---

diff --git a/content/browser/renderer_host/legacy_render_widget_host_win.cc b/content/browser/renderer_host/legacy_render_widget_host_win.cc
index 881c407..4a894ef 100644
--- a/content/browser/renderer_host/legacy_render_widget_host_win.cc
+++ b/content/browser/renderer_host/legacy_render_widget_host_win.cc
@@ -11,7 +11,6 @@
 
 #include "base/command_line.h"
 #include "base/win/win_util.h"
-#include "base/win/windows_version.h"
 #include "content/browser/accessibility/browser_accessibility_manager_win.h"
 #include "content/browser/accessibility/browser_accessibility_state_impl.h"
 #include "content/browser/accessibility/browser_accessibility_win.h"
@@ -134,18 +133,9 @@
   // heap-use-after-free crash (https://crbug.com/1194694).
   auto weak_ptr = weak_factory_.GetWeakPtr();
   RECT rect = {0};
-  DWORD window_ex_style = WS_EX_TRANSPARENT;
-  if (base::win::GetVersion() >= base::win::Version::WIN8) {
-    // For Windows 8 or greater set layered window (WS_EX_LAYERED) style to
-    // avoid black flash on first paint of LegacyRenderWidgetHostHWND.
-    // Also set WS_EX_NOREDIRECTIONBITMAP flag to avoid additional bitmap in
-    // layered window. Issue 1257540 (https://crbug.com/1257540).
-    window_ex_style |= WS_EX_LAYERED | WS_EX_NOREDIRECTIONBITMAP;
-  }
-
   Base::Create(parent, rect, L"Chrome Legacy Window",
                WS_CHILDWINDOW | WS_CLIPCHILDREN | WS_CLIPSIBLINGS,
-               window_ex_style);
+               WS_EX_TRANSPARENT);
   if (!weak_ptr) {
     // Base::Create() runs nested windows message loops that could end up
     // deleting `this`. Therefore, upon returning false here, `this` is already
